corehq/apps/geospatial/views.py

```diff
@@ -308,3 +308,19 @@ def _get_paginated_users_without_gps(domain, page, limit, query):
         'items': user_data,
         'total': paginator.count,
     }
+
+
+@require_GET
+@login_and_domain_required
+def get_users_with_gps(request, domain):
+    location_prop_name = get_geo_user_property(domain)
+    users = CommCareUser.by_domain(domain)
+    user_data = [
+        {
+            'id': user.user_id,
+            'username': user.raw_username,
+            'gps_point': user.pop_metadata(key=location_prop_name),
```

Comment:
Out of curiosity: I see other places are also doing `user.pop_metadata`, but why? Can't we just do `user.user_data[key]`?

Reply:
The reason I chose to use `pop_metadata` is because it is able to provide a default value if the key isn't found. In other places it seems to be used for removing unused fields from a user. 

As the name suggests, this will pop/remove the property from the user's metadata, but since we're not saving the user's metadata this should be fine. We could alternatively rather not remove it and instead do something like `user.metadata[location_prop_name] if location_prop_name in user.metadata else None` @SmittieC?

Reply:
Nah, not needed to change. Thanks for clarifying :+1:

Reply:
Not saying we should change this, just a quick thought that this could be risky. If in future someone modifies the code to save the user for any change, not knowing that we are removing a value from metadata in an earlier step, it could lead to metadata getting removed.
It's true that since this is a GET view, it should not do an update, but always better to not rely on assumptions.

Reply:
@mkangia That is a good point. After looking further into this, we can alternatively use `user.metadata.get(location_prop_name, '')` which will also provide us with a default if the property does not exist. The benefit of this is that we are simply reading from metadata, and not popping properties from it.

Since this removes any potential risk of losing user data, I'll make this change in the subsequent front-end PR for this work.

Reply:
Thank you!